### ISSUE TITLE
Prioritize moving view to the left margin when selecting multiple lines with `x`

### ISSUE THREAD

**Issue by kristian-clausal** [helix-editor/helix#7895]
This is just a minor quibble, but I didn't find anything about so, here goes.

I use an (around) 80-character width view, with a split terminal, so I understand I have less columns than is "hip" and "modern" in this day and age. As a consequence, I see this a lot:

In a file with overlong lines, selecting lines with `x` causes the cursor to jump around as it moves to the end of each lines. All good and expected, but I wish that when the cursor moves to a short line (less than width of viewable screen) that it would also scroll the view all the way to the left.

I understand the reasoning for why this works like this; when just scrolling with a cursor, you don't want the view to be wrenched all the way to the left without reason. However, I feel like there could be an exception for line-selecting with `x`, because in that case I am pretty sure 99% of the time you want to keep and eye on where you are in the text, and when the beginnings of lines are cut off (because the view doesn't return to the left margin unless you scroll through an empty line) that is difficult.

![example](https://github.com/helix-editor/helix/assets/93529179/9d615d01-f399-404c-8f2b-12b7c61a0c5c)

In this example image, the characters in capitals disappear out of view when selecting lines with `x`, even if the cursor moves towards left on a short line.

**Comment by pascalkuthe** [comment]
I don't think this is feasible. `x` is not aware where a selection came from. So there is no difference between `ghvgl` and `x'. So if you press `x' a second time it only checks what the current selection is (and whether it covers and entire line) so to implement this feature we would either need to:
* automatically scroll to the left if the current line fits entirely on the current screen (not a fan of this, this causes the view to realigning a ton and is not how any other editor behaves)
* track the last horizontal view offset in the selection itself. This adds a lof of overhead and complexity to a core editor component which I don't feel is justified by the usecase here.

So i don't think this should be implemented. Particulary because this can already be achieved much more easily in other ways:
* enable softwrap to avoid any horizontal scrolling
* perform your selection is visual mode with `j` and `k` and then extend to linebounds with `x`. This is generally the preferred way to work with line wise selection anyways (mirroring kak). Repeatetly hitting `x` is more of a shortcut when you want to select a small number of lines quickly.
